modules/data-entry/src/main/frontend/src/questionnaire/NumberQuestion.jsx

```diff
@@ -213,25 +213,31 @@ function NumberQuestion(props) {
 
       // Reject whitespace and non-numbers
       if (/^\s*$/.test(text) || isNaN(value)) {
-        return true;
+        return `The value${isRange ? 's' : ''} must be numeric`;
       }
     }
 
     // Test that it is within our min/max (if they are defined)
-    if ((typeof minValue !== 'undefined' && value < minValue) ||
-      (typeof maxValue !== 'undefined' && value > maxValue)) {
-      return true;
+    if ((typeof minValue !== 'undefined' && (isRange ? lowerLimit : value) < minValue) &&
+        (typeof maxValue !== 'undefined' && (isRange ? upperLimit : value) > maxValue)) {
+      return `The value${isRange ? 's' : ''} must be between ${minValue} and ${maxValue}`;
```

Comment:
This condition should only be possible if this is a range (since it needs to be both higher and lower than the limits) so can be simplified:
```suggestion
          if (isRange && typeof minValue !== 'undefined' && lowerLimit < minValue &&
            typeof maxValue !== 'undefined' && upperLimit > maxValue) {
            return `The values must be between ${minValue} and ${maxValue}`;
```